tensorflow/stream_executor/cuda/cuda_gpu_executor.cc

```diff
@@ -1129,6 +1129,30 @@ DeviceDescription *CUDAExecutor::PopulateDeviceDescription() const {
           CU_DEVICE_ATTRIBUTE_MAX_REGISTERS_PER_MULTIPROCESSOR, device_)
           .ValueOrDie());
 
+  const char* blank_ptx =
+      ".version 6.0\n"
+      ".target sm_30\n"
+      ".address_size 64\n"
+      "\n"
+      "        // .globl       _Z6ValAddPf\n"
+      ".visible .entry _Z6ValAddPf(\n"
+      ")\n"
+      "{\n"
+      "        ret;\n"
+      "}\n";
+  const char* kernel_name = "_Z6ValAddPf";
+
+  CUmodule blank_module;
+  CUfunction blank_function;
+  CUDADriver::LoadPtx(context_, blank_ptx, &blank_module);
+  CUDADriver::GetModuleFunction(context_, blank_module, kernel_name,
+                                &blank_function);
+
+  int bpc;
+  CUresult result = cuOccupancyMaxActiveBlocksPerMultiprocessor(
+      &bpc, blank_function, 1, (size_t)1);
```

Comment:
I'm not wild about the error checking in this function overall, but for this part I think we *do* want to add some reasonable error checking, because if e.g. the GPU is OOM it can fail to load a kernel

Reply:
Originally I tried to call CUDADriver::GetMaxOccupiedBlocksPerCore(), which wraps the driver API call with an error check, but the context activation was failing with an error stating that the context had been destroyed.  I called the driver function manually to get around this error, so one option would be to check the CUresult right after making this call. (As is done within the wrapped CUDADriver call)
```suggestion
      &bpc, blank_function, 1, (size_t)1);
      if (result != CUDA_SUCCESS) {
        //Handle failure
```
The next question is how we'd want to handle a failure here; one option is just to set the value to a fallback (like 32)

Reply:
> so one option would be to check the CUresult right after making this call. 

Yeah, I think we just need to be careful about checking these CUresults is all.  (Not sure how to do better than that.)

> The next question is how we'd want to handle a failure here; one option is just to set the value to a fallback (like 32)

Hmm...good question.  According to https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#features-and-technical-specifications, the true value is either 16 or 32.

I guess the question is, if we were to return -1, will the caller be likely to do a sensible thing?  If not, we should make up a value and return it.

Looking at the code in XLA, it sets blocks-per-sm equal to threads-per-block (?!), and if that fails, to 32.  So I guess returning 32 here is not so bad in comparison to that.  It would be nice to make it a somewhat-noisy failure, though, otherwise it's possible we'll never notice that it's going wrong (because 32 is usually the correct answer).  Since this code is only expected to run once per device, logging an ERROR is probably sufficient.